Move misc/vim-plugins to applications/editors/vim/plugins

AndersonTorres opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

This patch diminishes clutter in `pkgs/misc` and puts the plugins folder in a more intuitive place.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

AndersonTorres committed:
Move misc/vim-plugins to applications/editors/vim/plugins

AndersonTorres commented:
Oh an emacser modifying the vim scripts? I swear I am not sabotaging!

teto commented:
haven't check the PR itself but I agree with the principle. it would be more coherent with paths like ./pkgs/applications/editors/kakoune

AndersonTorres committed:
vim: update the updaters!

update.py and its companion update-shell.nix need to know where they are, else
they can't find the root default.nix of Nixpkgs. Because of this, I further
added a small piece of documentation about those path-dependent pieces of code.

AndersonTorres committed:
vim-plugin-names: updates

+ liuchengxu/space-vim
+ SpaceVim/SpaceVim

AndersonTorres commented:
Now the script list is updated

teto commented:
You need to update the .github/{CODEOWNERS/labeler.yaml} too. I dont mind testing and merging this but I would like some other vim-plugins maintainers' approval as this is disruptive to existing PRs.

AndersonTorres commented:
The code owners are @jonringer @softinio...

AndersonTorres commented:
@teto 

> You need to update the .github/{CODEOWNERS/labeler.yaml} too. 

Why exactly? Are those file path-related?

> I dont mind testing and merging this but I would like some other vim-plugins maintainers' approval as this is disruptive to existing PRs.

It can be merged after them.

AndersonTorres committed:
vim: Update .github/CODEOWNERS and .github/labeler.yml

AndersonTorres commented:
@jonringer

AndersonTorres committed:
vimPlugins: update

AndersonTorres committed:
vimPlugins: update
